mailcow/resource_mailbox.go

```diff
@@ -38,6 +40,21 @@ func resourceMailbox() *schema.Resource {
 				Required:    true,
 				ForceNew:    true,
 			},
+			"authsource": {
+				Type:        schema.TypeString,
+				Description: "Authentication source to use. One of: generic-oidc, mailcow, keycloak, ldap.",
+				Default:     "mailcow",
+				Optional:    true,
+				ValidateFunc: func(val any, key string) ([]string, []error) {
```

Comment:
please use [StringInSlice](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/v2@v2.36.1/helper/validation#StringInSlice)

Reply:
Sure, that's much easier...